int32 trainID
bool slowDown
